internal/integrationtest/board/board_test.go

```diff
@@ -558,3 +558,25 @@ func TestBoardSearchWithOutdatedCore(t *testing.T) {
 	// Installed version must be older than latest
 	require.True(t, installedVersion.LessThan(latestVersion))
 }
+
+func TestBoardListWithFailedBuiltinInstallation(t *testing.T) {
+	env, cli := integrationtest.CreateArduinoCLIWithEnvironment(t)
+	defer env.CleanUp()
+
+	_, _, err := cli.Run("core", "update-index")
+	require.NoError(t, err)
+
+	// board list to install builtin tools
+	_, _, err = cli.Run("board", "list")
+	require.NoError(t, err)
+
+	// remove files from serial-discovery directory to simulate a failed installation
+	serialDiscovery := cli.DataDir().Join("packages", "builtin", "tools", "serial-discovery", "1.4.0")
```

Comment:
Removing this specific version of the `serial-discovery` will result in a test failure when we'll release a new version of the tool in the future. A solution could be to remove the only release that should be inside the `serial-discovery/` directory